### ISSUE TITLE
Cannot edit/remove method comment if it's duplicated in the class

### ISSUE THREAD

**Issue by pierremorel-enactor** [javaparser/javaparser#1490]
I'm trying the following code

```
MethodDeclaration methodDeclaration = ...
// Cannot set a new javadoc comment
Javadoc javadoc = new Javadoc(JavadocDescription.parseText("Change Javadoc"));
methodDeclaration.setJavadocComment("", javadoc);

// Cannot remove Comment
methodDeclaration.removeComment();

// Cannot removeJavaDocComment
methodDeclaration.removeJavaDocComment();
```

on the following class:

```
public class MyClass {

  /**
   * Comment A
   */
  public void oneMethod() {
  }

  /**
   * Comment A
   */
  public void anotherMethod() {
  }

}
```

The issue seems to be that the javadoc comment is exactly the same for both methods.

Stack trace:
```
java.lang.IllegalStateException
	at com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter$Observer.concretePropertyChange(LexicalPreservingPrinter.java:179)
	at com.github.javaparser.ast.observer.PropagatingAstObserver.propertyChange(PropagatingAstObserver.java:62)
	at com.github.javaparser.ast.Node.lambda$notifyPropertyChange$3(Node.java:539)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.github.javaparser.ast.Node.notifyPropertyChange(Node.java:539)
	at com.github.javaparser.ast.Node.setComment(Node.java:248)
	at com.github.javaparser.ast.nodeTypes.NodeWithJavadoc.setJavadocComment(NodeWithJavadoc.java:71)
	at com.github.javaparser.ast.nodeTypes.NodeWithJavadoc.setJavadocComment(NodeWithJavadoc.java:76)
	at com.enactor.refactor.visitor.ReplaceInheritDocsAnnotationVisitor.visit(ReplaceInheritDocsAnnotationVisitor.java:58)
	at com.enactor.refactor.visitor.ReplaceInheritDocsAnnotationVisitor.visit(ReplaceInheritDocsAnnotationVisitor.java:1)
	at com.github.javaparser.ast.body.MethodDeclaration.accept(MethodDeclaration.java:120)
	at com.github.javaparser.ast.visitor.VoidVisitorAdapter.lambda$visit$29(VoidVisitorAdapter.java:170)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.github.javaparser.ast.NodeList.forEach(NodeList.java:267)
	at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:170)
	at com.github.javaparser.ast.body.ClassOrInterfaceDeclaration.accept(ClassOrInterfaceDeclaration.java:105)
	at com.github.javaparser.ast.visitor.VoidVisitorAdapter.lambda$visit$40(VoidVisitorAdapter.java:192)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.github.javaparser.ast.NodeList.forEach(NodeList.java:267)
	at com.github.javaparser.ast.visitor.VoidVisitorAdapter.visit(VoidVisitorAdapter.java:192)
```

**Comment by ThLeu** [comment]
It seems that this issue is already fixed, I added some tests in PR #1649 and they work fine.

**Comment by matozoid** [comment]
Good news, thanks for coming back to this :-)

**Comment by ThLeu** [comment]
hm...I just realised I might not have tested the same case in my PR as is mentioned here.
As I didn't use the LexcialPreservingPrinter in my tests and I would assume that changes only get propagated to the Printer if I use it, not in all cases :smile:.
I will have another look at it later.

**Comment by ThLeu** [comment]
I added a fix for this issue to the PR #1649 to be able to remove the comment (even the correct one of the two 😉).
The only issue that is left now is the indentation after the removal / replacement.
This can maybe be fixed with #1638.

PS: You could probably give the "Whitespace correction" its own label by now 😄

**Comment by ThLeu** [comment]
You can close this issue in my opinion, as it was fixed with PR #1649 
It doesn't work perfectly yet as mentioned there (whitespace), but the exception that occured is fixed.

**Comment by matozoid** [comment]
Good, I'll close it then.
